### ISSUE TITLE
What are valid commit types for `PrepareCommitMsg.ReplaceBranch.skipped_commit_types`?

### ISSUE THREAD

**Issue by 34875634567** [sds/overcommit#724]
I'm having trouble getting this hook to run only when 'manually' committing, and not during rebase/amend etc.

It looks like functionality to support this was introduced in https://github.com/sds/overcommit/commit/d46dbf0d918fa208c83fa801fe75c7c06e07c7d9, but it's not clear what the supported options are.

The following examples are given in [default.yaml](https://github.com/sds/overcommit/blob/b70cefcc992b109ca9905b4653a7cf2751f56fff/config/default.yml), are there others?
```
PrepareCommitMsg:
  ReplaceBranch:
    enabled: false
    description: 'Prepends the commit message with text based on the branch name'
    branch_pattern: '\A.*\w+[-_](\d+).*\z'
    replacement_text: '[#\1]'
    skipped_commit_types: ['message', 'template', 'merge', 'squash']
    on_fail: warn
```

**Comment by sds** [comment]
Based on what's written [here](https://github.com/sds/overcommit/blob/0491737438665563b888d6e6b0f262ba4a29eae1/lib/overcommit/hook_context/prepare_commit_msg.rb#L12-L17), like the commit types are 

* `message`
* `template`
* `merge`
* `squash`
* `commit`

I would ask the original author, @jethrodaniel, if they have anything to add about getting this working.

**Comment by jethrodaniel** [comment]
To avoid this running on `--amend`, just add `commit` to that list. Add `merge` to prevent it from running on `merge`s.

Actually, I've been meaning to find a way to keep this from running on `--amend` myself, but hadn't gotten around to actually doing anything about it... I had avoided adding `commit` to the `skipped_commit_types` because i thought that would prevent it from running on _any_ commits - turns out, the original [prepare-commit-msg doc](https://git-scm.com/docs/githooks#_prepare_commit_msg)'s wording is just confusing - `commit` is **not** the default, it only applies to `-c`, `-C`, and `--amend`.

I've clarified the hook documentation and added `commit` to that list by default in https://github.com/sds/overcommit/pull/725, since I don't think anyone wants to run this on `--amend` by default.
